libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineView.kt

```diff
@@ -452,33 +542,63 @@ class MapboxRouteLineView(options: MapboxRouteLineOptions) {
+                                it.sourceId,
+                                value.primaryRouteSource,
+                                newDataId
                             )
+                        }
+                        sourceToFeatureMap[it] = RouteLineFeatureId(routeId)
                     }
                     sourceLayerMap.keys.filter { it != primarySourceKey }
                         .forEachIndexed { index, routeLineSourceKey ->
                             if (index < value.alternativeRouteSourceSources.size) {
-                                updateSource(
-                                    style,
+                                val newDataId = dataIdHolder
+                                    .incrementDataId(routeLineSourceKey.sourceId)
+                                updateSourceCommands.add {
+                                    updateSource(
+                                        style,
+                                        routeLineSourceKey.sourceId,
+                                        value.alternativeRouteSourceSources[index],
+                                        newDataId
+                                    )
+                                }
+                                val routeId = value.alternativeRouteSourceSources[index]
+                                    .features()
+                                    ?.firstOrNull()
+                                    ?.id()
+                                expectedRoutesData.addClearedRoute(
                                     routeLineSourceKey.sourceId,
-                                    value.alternativeRouteSourceSources[index]
+                                    newDataId,
+                                    sourceToFeatureMap[routeLineSourceKey]?.id()
                                 )
-                                sourceToFeatureMap[routeLineSourceKey] =
-                                    RouteLineFeatureId(
-                                        value.alternativeRouteSourceSources[index]
-                                            .features()
-                                            ?.firstOrNull()
-                                            ?.id()
-                                    )
+                                expectedRoutesData.addRenderedRoute(
+                                    routeLineSourceKey.sourceId,
+                                    newDataId,
+                                    routeId
+                                )
+                                sourceToFeatureMap[routeLineSourceKey] = RouteLineFeatureId(routeId)
                             }
                         }
+                    if (callback != null) {
+                        routesExpector.expectRoutes(
+                            expectedRoutesData.allRenderedRouteIds,
+                            expectedRoutesData.allClearedRouteIds,
+                            expectedRoutesData,
```

Comment:
We're already passing the whole `expectedRoutesData`, why do we need to pass IDs separately?